noir-projects/noir-contracts/contracts/token_bridge_contract/src/main.nr

```diff
@@ -135,31 +131,17 @@ pub contract TokenBridge {
         caller_on_l1: EthAddress, // ethereum address that can call this function on the L1 portal (0x0 if anyone can call)
         nonce: Field, // nonce used in the approval message by `msg.sender` to let bridge burn their tokens on L2
     ) {
+        let config = storage.config.read();
+
+        // Assert that user provided token address is same as seen in storage.
+        assert_eq(config.token, token, "Token address is not the same as seen in storage");
+
         // Send an L2 to L1 message
         let content = get_withdraw_content_hash(recipient, amount, caller_on_l1);
-        context.message_portal(storage.portal_address.read(), content);
+        context.message_portal(config.portal, content);
 
-        // docs:start:call_assert_token_is_same
-        // Assert that user provided token address is same as seen in storage.
-        TokenBridge::at(context.this_address())._assert_token_is_same(token).enqueue(&mut context);
```

Comment:
I think this was done before we were able to read public storage from private. I just replaced that with the inlined assert above.